Ollivanders/src/net/pottercraft/ollivanders2/spell/APPARATE.java

```diff
@@ -298,6 +305,30 @@ private boolean canApparateTo(@NotNull Location destination)
         return true;
     }
 
+    /**
+     * Determine if the distance a player is trying to apparate exceeds the max apparate distance in the X or Z coordinates.
+     *
+     * @param fromLoc the location they are apparating from
+     * @param toLoc the location they are apparating to
+     * @return true if the locations are less than maxApparateDistance apart, false otherwise
+     */
+    private boolean exceedsMaxDistance (@NotNull Location fromLoc, @NotNull Location toLoc)
+    {
+        p.getServer().getLogger().info("checking max apparate distance " + maxApparateDistance);
+        // value less 1 means there is no limit
+        if (maxApparateDistance < 1)
+            return false;
+
+        // ensure the player is not trying to go more than maxApparateDistance blocks in the x and z directions
+        if (Math.abs(fromLoc.getX() - toLoc.getX()) > maxApparateDistance)
```

Comment:
We need to check only the X and Z coordinates because Y is limited by MC itself.